pkg/pillar/types/assignableadapters.go

```diff
@@ -460,6 +524,30 @@ func (aa *AssignableAdapters) LookupIoBundleIfName(ifname string) *IoBundle {
 	return nil
 }
 
+type ownParentError struct{}
+
+func (o ownParentError) Error() string {
+	return "IOBundle cannot be it's own parent"
+}
+
+type parentAssigngrpMismatchError struct{}
+
+func (p parentAssigngrpMismatchError) Error() string {
+	return "IOBundle with parentassigngrp mismatch found"
+}
+
+type emptyAssigngrpWithParentError struct{}
+
+func (e emptyAssigngrpWithParentError) Error() string {
+	return "IOBundle with empty assigngrp cannot have a parent"
+}
+
+type cycleDetectedError struct{}
+
+func (c cycleDetectedError) Error() string {
+	return "Cycle detected, please check provided parentassigngrp/assigngrp"
+}
+
 // CheckParentAssigngrp finds dependency loops between ioBundles
```

Comment:
Can you update the comment to say that it will re-evaluate/update (clear and set any new) errors relating to the parent assigngrp. Current comment implies it doesn't clear any.

Reply:
now: `// CheckParentAssigngrp finds dependency loops between ioBundles and sets/clears the error`